### PR TITLE
refactor: option for classes to create instances from factory path

### PR DESCRIPTION
**Pull request by ludwiktrammer**

### CODE REVIEW

File: packages/ragbits-core/src/ragbits/core/utils/config_handling.py [hunk 1]
```diff
@@ -13,39 +13,39 @@ class InvalidConfigError(Exception):
     """
 
 
-def get_cls_from_config(cls_path: str, default_module: ModuleType | None) -> Any:  # noqa: ANN401
+def import_by_path(path: str, default_module: ModuleType | None) -> Any:  # noqa: ANN401
```
Comment (ludwiktrammer) [inline packages/ragbits-core/src/ragbits/core/utils/config_handling.py:16]: Note: those changes don't modify the behavior of this function - they just better describe the behavior that it had all along.

File: packages/ragbits-core/tests/unit/utils/pyproject/testprojects/bad_factory_project/pyproject.toml [hunk 2]
```diff
@@ -2,6 +2,6 @@
 name = "bad_factory_project"
 
 [tool.ragbits.core.default_llm_factories]
-non_existing = "ragbits.core.llms.factory.simple_litellm_factory"
-vision = "ragbits.core.llms.factory.simple_litellm_vision_factory"
-structured_output = "ragbits.core.llms.factory.simple_litellm_vision_factory"
\ No newline at end of file
+non_existing = "ragbits.core.llms.factory:simple_litellm_factory"
+vision = "ragbits.core.llms.factory:simple_litellm_vision_factory"
+structured_output = "ragbits.core.llms.factory:simple_litellm_vision_factory"
```
Comment (micpst) [inline packages/ragbits-core/tests/unit/utils/pyproject/testprojects/bad_factory_project/pyproject.toml:7]: new line

File: packages/ragbits-core/tests/unit/utils/pyproject/testprojects/factory_project/pyproject.toml [hunk 3]
```diff
@@ -2,6 +2,6 @@
 name = "factory_project"
 
 [tool.ragbits.core.default_llm_factories]
-text = "ragbits.core.llms.factory.simple_litellm_factory"
-vision = "ragbits.core.llms.factory.simple_litellm_vision_factory"
-structured_output = "ragbits.core.llms.factory.simple_litellm_vision_factory"
\ No newline at end of file
+text = "ragbits.core.llms.factory:simple_litellm_factory"
+vision = "ragbits.core.llms.factory:simple_litellm_vision_factory"
+structured_output = "ragbits.core.llms.factory:simple_litellm_vision_factory"
```
Comment (micpst) [inline packages/ragbits-core/tests/unit/utils/pyproject/testprojects/factory_project/pyproject.toml:7]: ditto

File: packages/ragbits-core/tests/unit/utils/pyproject/testprojects/bad_factory_project/pyproject.toml (hunk 2 above)
Comment (micpst) [inline packages/ragbits-core/tests/unit/utils/pyproject/testprojects/bad_factory_project/pyproject.toml:7]: new line

File: packages/ragbits-core/tests/unit/utils/pyproject/testprojects/factory_project/pyproject.toml (hunk 3 above)
Comment (micpst) [inline packages/ragbits-core/tests/unit/utils/pyproject/testprojects/factory_project/pyproject.toml:7]: ditto

**Review by mhordynski — APPROVED** [review]

File: packages/ragbits-core/tests/unit/utils/pyproject/testprojects/bad_factory_project/pyproject.toml (hunk 2 above)
Comment (ludwiktrammer) [reply packages/ragbits-core/tests/unit/utils/pyproject/testprojects/bad_factory_project/pyproject.toml:7]: This PR didn't change anything in that regard. I just mechanically changed the format for giving LLM factories in the entire project. The end of the file was left exactly as it was before.

Comment (micpst) [reply packages/ragbits-core/tests/unit/utils/pyproject/testprojects/bad_factory_project/pyproject.toml:7]: ok, this is not related to the change, but we can add this new line at the end of the file to remove this annoying gh warning

Comment (ludwiktrammer) [reply packages/ragbits-core/tests/unit/utils/pyproject/testprojects/bad_factory_project/pyproject.toml:7]: By "annoying gh warning" you mean the red circle that symbolise end of file without new line? Ok, I'll do it, but if that's something we are serious about, we should just configure a linter instead

Comment (ludwiktrammer) [reply packages/ragbits-core/tests/unit/utils/pyproject/testprojects/bad_factory_project/pyproject.toml]: Done

Comment (micpst) [reply packages/ragbits-core/tests/unit/utils/pyproject/testprojects/bad_factory_project/pyproject.toml]: totally agree with the linter part

**Review by micpst — APPROVED** [review]

**Pull request merged**